ggshield/verticals/secret/secret_scan_collection.py

```diff
@@ -19,6 +22,40 @@ class Result(NamedTuple):
     file: Scannable  # filename that was scanned
     scan: ScanResult  # Result of content scan
 
+    def enrich_matches(self):
+        content = self.content
+        is_patch = self.filemode != Filemode.FILE
+        lines = get_lines_from_content(content, self.filemode)
+        for policy_break in self.scan.policy_breaks:
+            res = []
+            for match in policy_break.matches:
+                if match.index_start is None or match.index_end is None:
+                    res.append(match)
+                    continue
+                span = MatchSpan.from_match(match, lines, is_patch)
+                line_start = lines[span.line_index_start]
+                line_end = lines[span.line_index_end]
+                line_index_start = line_start.pre_index or line_start.post_index
+                line_index_end = line_end.pre_index or line_end.post_index
+                assert line_index_start is not None and line_index_end is not None
+                line_index_start += int(is_patch) - 1  # convert to 0-based
+                line_index_end += int(is_patch) - 1
+                res.append(
+                    ExtendedMatch(
+                        match=match.match,
+                        match_type=match.match_type,
+                        index_start=span.column_index_start,
+                        index_end=span.column_index_end,
+                        line_start=line_index_start,
+                        line_end=line_index_end,
+                        pre_line_start=line_start.pre_index,
+                        post_line_start=line_start.post_index,
+                        pre_line_end=line_end.pre_index,
+                        post_line_end=line_end.post_index,
+                    )
+                )
+            policy_break.matches = res
```

Comment:
If you introduce a `def enrich_match(Match, List[Lines]) -> Match:` function then you can simplify these loops to something like this:

```python
for policy_break in self.scan.policy_breaks:
   policy_break.matches = [enrich_match(lines, m) for m in policy_break.matches]
```